nixos/modules/services/misc/metabase.nix

```diff
@@ -0,0 +1,117 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.metabase;
+
+  inherit (lib) mkEnableOption mkIf mkOption;
+  inherit (lib) optional optionalAttrs optionalString types;
+
+  dataDir = "/var/lib/metabase";
+
+in {
+
+  options = {
+
+    services.metabase = {
+      enable = mkEnableOption "Metabase service";
+
+      listen = {
+        ip = mkOption {
+          type = types.str;
+          default = "0.0.0.0";
+          description = ''
+            IP address that Metabase should listen on.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 3000;
+          description = ''
+            Listen port for Metabase.
+					'';
+        };
+      };
+
+			ssl = {
+				enable = mkOption {
+					type = types.bool;
+					default = false;
+					description = ''
+            Whether to enable SSL (https) support.
+					'';
+				};
+
+				port = mkOption {
+					type = types.port;
+					default = 8443;
+					description = ''
+            Listen port over SSL (https) for Metabase.
+					'';
+				};
+
+				keystore = mkOption {
+          type = types.nullOr types.path;
+          default = "${dataDir}/metabase.jks";
+          example = "/etc/secrets/keystore.jks";
+          description = ''
+            Java KeyStore file containing the certificates.
```

Comment:
Are there any upstream docs for this you could link to?

Reply:
Upstream points to https://www.digitalocean.com/community/tutorials/java-keytool-essentials-working-with-java-keystores but I did not want to include it as it's a standard Java Keystore.

Reply:
I think this would be great to link to, especially when upstream links to it too

Reply:
Ok, added link in the description.